### ISSUE TITLE
关于训练损失函数

### ISSUE THREAD

**Issue by Qiqigeww** [hq-deng/AnoVL#4]
作者，你好，我看到代码里面损失函数部分如下：
def loss_func(pred):
    soft_loss = -pred[0]*pred[0].log()
    mask = torch.zeros(pred[1:].shape).to(pred[1:].device)
    mask[...,1] = 1
    hard_loss = -mask*pred[1:].log()-(1-mask)*pred[0].log()
    soft_loss = soft_loss.sum(-1).mean()
    hard_loss = hard_loss.sum(-1).mean()
    #alpha=0.9
    return 1*soft_loss+0.5*hard_loss
而论文里面的损失函数写作如下图所示：
<img width="615" alt="image" src="https://github.com/hq-deng/AnoVL/assets/67260396/e77943b5-754a-4dd7-a6ac-4f238c24513d">
感觉难以对应起来，怀疑代码这里写的有点问题，能解释一下吗？

**Comment by hq-deng** [comment]
Hello, there are some mistakes in our paper, we will update a revised version later. Thanks.

**Comment by hq-deng** [comment]
We have deduced the equations again, and there is no mistake in the equation. Both of the objective functions can be treated as cross-entropy, and we omit the y(mask)=0 item in the function. If there is any question, please let me know.
